eth/protocols/snap/sync.go

```diff
@@ -1854,6 +1915,9 @@ func (s *Syncer) processTrienodeHealResponse(res *trienodeHealResponse) {
 	if err := batch.Write(); err != nil {
 		log.Crit("Failed to persist healing data", "err", err)
 	}
+	if err := batch.Replay(newStateWriter(res, s)); err != nil {
```

Comment:
Clever!

Reply:
Actually it;s totally wrong :) I have fixed it.